### PR TITLE
[Refactor] RemoteParticipantsManager update

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
## Purpose

* Decouple RemoteParticipantManager from CallSDKWrapper
* Removed CommunicationIdetifierExtension

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## What to Check

- Code can be built
- Tests pass

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AvatarViewManager.swift [hunk 1]
```diff
@@ -60,8 +60,8 @@ class AvatarViewManager: AvatarViewManagerProtocol, ObservableObject {
              for identifier: CommunicationIdentifier,
              completionHandler: ((Result<Void, SetParticipantViewDataError>) -> Void)? = nil) {
         let participantsList = store.state.remoteParticipantsState.participantInfoList
-        guard let idStringValue = identifier.stringValue,
-              participantsList.contains(where: { $0.userIdentifier == idStringValue })
+        let idStringValue = identifier.rawId
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AvatarViewManager.swift:63]: nitpick: I'm not sure there is much value in keeping this `idStringValue`

**Review by JoshuaLai — APPROVED** [review]

**Review by emlynmac — APPROVED** [review]
This PR makes me happy :)

**Comment by JoshuaLai** [comment]
Oh can we make sure Android is doing an equivalent change on their side? 
@vhuseinova-msft @iaulakh @pavelprystinka

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AvatarViewManager.swift (hunk 1 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/AvatarViewManager.swift:63]: it's used in a couple of other places in this func so I just left it as it is

**Comment by vhuseinova-msft** [comment]
> Oh can we make sure Android is doing an equivalent change on their side?
> @vhuseinova-msft @iaulakh @pavelprystinka

Good catch @JoshuaLai! @iaulakh has created a task for this

**Pull request merged**